File: mod/ondb/lib/js/get.js
import { Ondb } from './ondb_core.js';

let eid = '';
for (let i = 2; i < Bun.argv.length; i++) {
  const a = Bun.argv[i];
  if (a.startsWith('--id=')) eid = a.slice(5);
}

const db = new Ondb();
const input = await Bun.stdin.text();
db.load(input);

const e = db.get(eid);
if (e) {
  const out = [
    `id\t${e.id}`,
    `type\t${e.type}`,
    `ctime\t${e.ctime}`,
    `mtime\t${e.mtime}`
  ];
  for (const k in e.props) {
    out.push(`prop\t${k}=${e.props[k]}`);
  }
  process.stdout.write(out.join('\n') + '\n');
}

File: mod/ondb/lib/py/get.py
import sys, os
sys.path.insert(0, os.path.dirname(__file__))
from ondb_core import Ondb

eid = ''
for a in sys.argv[1:]:
    if a.startswith('--id='): eid = a.split('=', 1)[1]

db = Ondb()
db.load(sys.stdin)
e = db.get(eid)
if e:
    print(f"id\t{e['id']}")
    print(f"type\t{e['type']}")
    print(f"ctime\t{e['ctime']}")
    print(f"mtime\t{e['mtime']}")
    for k, v in e['props'].items():
        print(f"prop\t{k}={v}")
